homeassistant/components/hydrawise/coordinator.py

```diff
@@ -0,0 +1,29 @@
+"""DataUpdateCoordinator for the Hydrawise integration."""
+
+from __future__ import annotations
+
+from datetime import timedelta
+
+from hydrawiser.core import Hydrawiser
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN, LOGGER
+
+
+class HydrawiseDataUpdateCoordinator(DataUpdateCoordinator):
+    """The Hydrawise Data Update Coordinator."""
+
+    def __init__(
+        self, hass: HomeAssistant, api: Hydrawiser, scan_interval: timedelta
+    ) -> None:
+        """Initialize HydrawiseDataUpdateCoordinator."""
+        super().__init__(hass, LOGGER, name=DOMAIN, update_interval=scan_interval)
+        self.api = api
+
+    async def _async_update_data(self):
+        """Fetch the latest data from Hydrawise."""
+        result = await self.hass.async_add_executor_job(self.api.update_controller_info)
```

Comment:
Doesn't look like we use the coordinator to fetch any data?

Reply:
It does. The `update_controller_info()` method caches the data inside the `api` object. (Not how I would have designed the API, but it's what I'm working with)

I can pull out the relevant data if you really want me to, but I also want to replace the API eventually with https://github.com/dknowles2/pydrawise/ which uses the newer GraphQL version of the Hydrawise service. The data returned by that API will be vastly different than this one, so I'd rather not create more work for myself later.